builder/dockerfile/copy_windows.go

```diff
@@ -5,17 +5,63 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/Microsoft/go-winio"
 	"github.com/docker/docker/pkg/idtools"
+	"github.com/docker/docker/pkg/system"
+	"golang.org/x/sys/windows"
 )
 
 var pathBlacklist = map[string]bool{
 	"c:\\":        true,
 	"c:\\windows": true,
 }
 
-func fixPermissions(source, destination string, rootIDs idtools.IDPair, overrideSkip bool) error {
-	// chown is not supported on Windows
-	return nil
+func fixPermissions(source, destination string, identity idtools.Identity, overrideSkip bool) error {
+	if identity.SID == "" {
+		return nil
+	}
+
+	privileges := []string{winio.SeRestorePrivilege, system.SeTakeOwnershipPrivilege}
+
+	err := winio.EnableProcessPrivileges(privileges)
```

Comment:
Sorry, I've been OOF for what seems like forever, so haven't had time to look at this before. This here though makes me extremely nervous. We shouldn't be doing this in the daemon as it's global and could be handling other user requests. That's exactly why we re-exec the layerwriter in the Windows graphdriver to start another process which has these privileges.

Reply:
That's a good point, I'll change this to use re-exec instead.